src/matrix/net/common.ts

```diff
@@ -27,9 +35,10 @@ export function encodeQueryParams(queryParams) {
         .join("&");
 }
 
-export function encodeBody(body) {
-    if (body.nativeBlob && body.mimeType) {
-        const blob = body;
+export function encodeBody(body: Record<string, any>): IEncodedBody {
```

Comment:
nit: [`JSON.stringify()`](https://github.com/microsoft/TypeScript/blob/cec2fda9a53620dc545a2c4d7b0156446ab145b4/lib/lib.es5.d.ts#L1066) seems to use `any` as the object type. `Record` seems more for objects with [uniform value types](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type) AIUI.

```suggestion
export function encodeBody(body: BlobHandle | any): IEncodedBody {
```

`any` already includes `BlobHandle`, but it's a bit clearer we can pass in a blob as well.

Reply:
The problem with `any` is that it also allows primitive values (number, boolean ...) to be passed. I think what we want here might be `BlobHandle | object` since we don't access any keys in body?